From b091e70df181417ba7d3fff48300c2cbc093414c Mon Sep 17 00:00:00 2001
From: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
Date: Tue, 30 Mar 2021 16:48:25 +0200
Subject: [PATCH] ovs-vswitchd.service: add sandboxing options

Add sandboxing options to ovs-vswitchd.service. The options set are the
following:
* PrivateTmp: secure access to temporary files of the process
* NoNewPrivileges: ensure the process and children can not gain new
  privileges
* ProtectSystem=strict: mount all filesystem read-only except /dev,
  /proc and /sys
* ReadWritePaths: list of directories the process can write to
* ProtectKernelModules: prevent the service from loading kernel modules
* ProtectKernelTunables: make /proc/sys, /sys and other kernel tunables
  read-only
* ProtectControlGroups: make /sys/fs/cgroups read-only
* RestrictSUIDGID: prevent the process from setting SUID/GID bits
* RestrictNamespaces: restrict access to the list of Linux namespaces
* CapabilityBoundingSet: restrict the capabilities available to the
  running process
* PrivateDevices: prevent access to /dev

Signed-off-by: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
---
 ...usr_lib_systemd_system_ovs-vswitchd.service.in | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
index b7a89b334..b1e03c568 100644
--- a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
+++ b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
@@ -31,3 +31,18 @@ ExecReload=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server \
           ${OVS_USER_OPT} \
           restart $OPTIONS
 TimeoutSec=300
+
+# Sandboxing
+PrivateTmp=yes
+NoNewPrivileges=true
+ProtectSystem=strict
+ReadWritePaths=/run /var/run/openvswitch /var/log/openvswitch
+ProtectKernelModules=yes
+ProtectKernelTunables=yes
+ProtectControlGroups=yes
+RestrictSUIDGID=true
+RestrictNamespaces=pid user cgroup
+# CAP_IPC_LOCK is required for mlockall calls
+CapabilityBoundingSet=CAP_SYS_NICE CAP_NET_ADMIN CAP_IPC_LOCK
+# Access to devices is required for DPDK in some conf
+# PrivateDevices=yes
-- 
2.25.1

